src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -26,5 +68,61 @@ private static bool GetSwitchValue(string switchName, string envVariable)
 
             return ret;
         }
+
+        private static bool TryGetStringValue(string switchName, string envVariable, [NotNullWhen(true)] out string? value)
+        {
+            value = AppContext.GetData(switchName) as string;
+            if (string.IsNullOrEmpty(value))
+            {
+                value = Environment.GetEnvironmentVariable(envVariable);
+                if (string.IsNullOrEmpty(value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static string CreateLibraryName(ReadOnlySpan<char> baseName, ReadOnlySpan<char> suffix, ReadOnlySpan<char> extension, ReadOnlySpan<char> version, bool versionAtEnd = false)
+        {
+            int length = baseName.Length + suffix.Length + version.Length + extension.Length;
+
+            // We validate that suffix and version are not larger than 53 characters.
```

Comment:
Implementing this using `string.Concat` would be much more readable, and efficient too.

Reply:
Yup.  We have a four-span-overload of string.Concat, so this would end up being something like:
```C#
private static string CreateLibraryName(ReadOnlySpan<char> baseName, ReadOnlySpan<char> suffix, ReadOnlySpan<char> extension, ReadOnlySpan<char> version, bool versionAtEnd = false) =>
    string.Concat(baseName, suffix, versionAtEnd ? extension : version, versionAtEnd ? version : extension);
```

Reply:
Thanks for the suggestion.